game.py

```diff
@@ -0,0 +1,908 @@
+        board_even = False
+    else:
+        board_even = True
+   
+    cyborg = False
+        
+    global score_count
+    global score_find
+    global total_score_count
+    total_score_count = 0
+    game_state.print_board()
+   
+    another_move = True
+    while another_move:
+        if not cyborg:
+            cyborg = get_cyborg()
+            if cyborg:
+                cyborg_look_forward = get_look_forward()
+                cyborg_sort_depth = get_sort_depth()
+                score_find = 9876
+        if cyborg:
+            best_move_and_score = get_computer_move(game_state)
+            # print ('best move and score: ', best_move_and_score, 'Score count : ', score_count)
+            best_move = best_move_and_score[0]
+            game_state.make_move(best_move)
+        else:    
+            if  human_to_play():
+                while get_hint():
+                    # score_find = 9876
+                    # score_find = get_score_find()
+                    best_move_and_score = get_computer_move(game_state)
+                    best_move = best_move_and_score[0]
+                    # print('Suggested move row : ', best_move[0] + 1, 'column : ', best_move[1] + 1)
+                    print('Suggested move column : ', best_move[1] + 1)
+            
+                move = get_human_move(game_state)
+                # game_state = game_state.make_move(move)
+                game_state.make_move(move)
+            else:
+                # score_find = 9876
+                # score_find = get_score_find()
+            
+                best_move_and_score = get_computer_move(game_state)
+                # print ('best move and score: ', best_move_and_score, 'Score count : ', score_count)
+                best_move = best_move_and_score[0]
+                # game_state = game_state.make_move(best_move)
+                game_state.make_move(best_move)
+                 
+        game_state.print_board()
+        
+        if not cyborg:
+            while undo_move():
+                game_state.unmake_move()
+                game_state.print_board()
+                #if game_state.get_last_move() == None:
+                if game_state.get_number_of_moves() == 0:
+                    break
+                
+        game_end_win = game_state.game_end()
+        
+        if game_end_win[0]:
+            print('End of Game')
+            if game_end_win[1]:
+                if game_state.get_player_next() == 1:
+                    print('Winner is Black')
+                else:
+                    print('Winner is White')
+            else: 
+                print('Game is Drawn')
+            print('Time for game : ', round(time.time()- game_start_time, 1), ' seconds')
+            print('Total score count : ', total_score_count)
+            break
+
+        if not cyborg:
+            save_game(game_state)
+            another_move = move_more()
+
+     
+if __name__ == '__main__':
+    main()

```

Comment:
I highly encourage you at this time to start splitting out this into multiple files. This will force you to grapple with modules and imports, which is a learning curve, but is very helpful and not too hard once you get the knack. This file-length is unwieldly.

It also has a great in-built forcing function of making you think about naming and sub-categories for you functions. Maybe you'll have one for user-interaction, another for the AI, and then another for all of your classes. These can then be further split-down if you'd like. It will help you navigate/read your code too, especially if you are using a good editor/IDE (I recommend PyCharm) so you aren't just getting good at scrolling up/down several 100 lines constantly